src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -638,6 +648,9 @@ public function setAlias($alias, $id)
         }
 
         unset($this->definitions[$alias]);
+        if ($alias !== $caseSensitiveAlias) {
+            $this->caseSensitiveIds[$alias] = $caseSensitiveAlias;
```

Comment:
Do we actually need the entry for aliases or could we just remove existing entries here?